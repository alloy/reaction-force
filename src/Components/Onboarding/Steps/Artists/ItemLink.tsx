import * as React from "react"
import { createFragmentContainer, graphql } from "react-relay/compat"
import styled, { StyledFunction } from "styled-components"

import { fadeIn, fadeOut } from "../../../../Assets/Animations"
import * as fonts from "../../../../Assets/Fonts"

import Icon from "../../../Icon"

import UpdateArtistFollowMutation from "./Mutations/UpdateArtistFollow"

const anchor: StyledFunction<State & React.HTMLProps<HTMLInputElement>> = styled.a
const Link = anchor`
  display: flex;
  font-size: 14px;
  color: black;
  text-decoration: none;
  text-transform: uppercase;
  font-family: ${fonts.primary.fontFamily};
  border-top: 1px solid #e5e5e5;
  &:hover {
    background-color: #f8f8f8;
  }
  ${props => (props.fadeIn ? fadeIn : null)}
  ${props => (props.fadeOut ? fadeOut : null)}
`

const Avatar = styled.img`
  padding: 10px 15px 10px 10px;
`

const FullWidthCol = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
`

const Col = styled.div`
  display: flex;
  align-items: center;
`

export interface State {
  fadeIn: boolean
  fadeOut: boolean
  artist_id: string
}

type Props = React.HTMLProps<HTMLAnchorElement> & RelayProps

class ItemLink extends React.Component<Props, State> {
  state = {
    fadeIn: false,
    fadeOut: false,
    artist_id: this.props.artist.id,
  }

  followArtist() {
    // const mutation = new UpdateArtistFollowMutation(this.state)

    // Relay.Store.commitUpdate(mutation, {
    //   onFailure: this.followArtistFailed,
    //   onSuccess: response => {
    //     // do something
    //   },
    // })
    console.log("follow artist!")
  }

  followArtistFailed() {
    null
  }

  onClick() {
    this.followArtist()
  }

  render() {
    return (
      <Link
        onClick={this.onClick.bind(this)}
        fadeIn={this.state.fadeIn}
        fadeOut={this.state.fadeOut}
        artist_id={this.props.artist.id}
      >
        <Col>
          {<Avatar src={this.props.artist.image ? this.props.artist.image.cropped.url : ""} width={50} height={50} />}
        </Col>
        <FullWidthCol>{this.props.artist.name}</FullWidthCol>
        <Col>
          <Icon name="follow-circle" color="black" fontSize="39px" />
        </Col>
      </Link>
    )
  }
}

export default createFragmentContainer(
  ItemLink,
  graphql`
    fragment ItemLink_artist on Artist {
      id
      name
      image {
        cropped(width: 100, height: 100) {
          url
        }
      }
    }
  `
)

export interface RelayProps {
  artist: {
    id: string | null
    name: string | null
    image: {
      cropped: {
        url: string | null
      } | null
    } | null
  }
}
