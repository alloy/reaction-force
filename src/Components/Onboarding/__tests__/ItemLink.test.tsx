import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import ItemLink from "../ItemLink"

describe("ItemLink", () => {
  describe("snapshot", () => {
    it("renders an item link", () => {
      const component = renderer
        .create(
          <ItemLink id="123" name="Andy Warhol" image_url="https://image.jpg" />
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
