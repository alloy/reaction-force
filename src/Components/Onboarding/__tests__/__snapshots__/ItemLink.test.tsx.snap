// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemLink snapshot renders an item link 1`] = `
.c6 {
  font-family: artsy-icons;
  color: white;
  font-size: 11.55px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c5 {
  width: 21px;
  height: 21px;
  background-color: black;
  border-radius: 50%;
  vertical-align: text-bottom;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-item: center;
  -webkit-box-align-item: center;
  -ms-flex-align-item: center;
  align-item: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  color: black;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c0:hover {
  background-color: #f8f8f8;
  cursor: pointer;
}

.c2 {
  margin: 10px 15px 10px 10px;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  width: 50px;
  text-align: center;
}

<a
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <img
      className="c2"
      height={50}
      src="https://image.jpg"
      width={50}
    />
  </div>
  <div
    className="c3"
  >
    Andy Warhol
  </div>
  <div
    className="c1"
  >
    <div
      className="c4"
    >
      <div
        className="c5"
        color="black"
        fontSize="21px"
        name="close"
        style={
          Object {
            "transform": "rotate(45deg)",
          }
        }
      >
        <div
          className="c6"
          onClick={undefined}
          style={
            Object {
              "alignSelf": "center",
              "margin": 0,
            }
          }
        >
          î˜”
        </div>
      </div>
    </div>
  </div>
</a>
`;
